### ISSUE TITLE
Using @NaturalId for cache

### ISSUE THREAD

**Issue by jhhong0509** [spring-projects/spring-data-jpa#2315]
I'm using `@NaturalId` annotation in my `User.email`.
I expected that my `findByEmail(String email)` would enable to search in first level cache or second level cache.
But in log, cache does not work as I think.
Every `findByEmail` method were access DB.

I used `bySimpleNaturalId()` for my purpose, but it's very hassle.
Please make `@NaturalId` access to cache enable

**Comment by schauder** [comment]
This is an interesting proposal. Lets wait until it is actually part of JPA: https://github.com/eclipse-ee4j/jpa-api/issues/320

We try to avoid implementation specifics in Spring Data JPA.

**Comment by schauder** [comment]
Yes, because Spring Data JPA certainly won't get into caching itself.

**Comment by nightswimmings** [comment]
I meant caching it though hinting the JPA Provider. 
Idk, it's just that it feels so antifunctional to use findByNaturalId() when findByEmail(), findByISBN() or findByName() is so readable :,(

**Comment by ratsam** [comment]
I know this is an old issue, but for anyone arriving here from Google search, this project might help: https://github.com/pocketcombats/spring-data-naturalid
